packages/common/src/env.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from "zod";
+
+interface Options {
+  required: boolean;
+  defaultValue?: string;
+}
+
+const defaultOptions = {
+  required: true,
+  defaultValue: undefined,
+};
+
+export function getEnv(
```

Comment:
Not very convinced about this, seems a step back from just using zod since it duplicates `optional` and `default` functionality but prevents using the rest (e.g. coercion). An indirection more than an abstraction.

If we're going toward the ideal situation of parsing env just once, just in one place, then passing the validated object around, I'd rather avoid building intermediate utilities.

By the way, we've talked about that end goal as something that needs quite a bit of work, but if we're content with having it at the module level for now rather than being passed down via React context, it could be as easy as:

```ts
// config.ts

let cached: Config | null = null

const getConfig = (env = process.env): Config => {
  if (cached === null) {
    cached = {
      val: z.string().default('...').parse(env.SOME_VALUE),
      val2: z.string().optional().parse(env.SOME_OTHER_VALUE)
    }
  }
  return cached
}

// foo.tsx

import { getConfig } from 'common/config'

const Comp: React.FC = () => {
  const { val, val2 } = getConfig();
  // ...
}
```

That way, `config.ts` would be the only place that deals with `zod` (at least for this purpose) and the only place where reading from the environment can fail; everything else can assume the object to be valid.

Reply:
good idea! I didn't want to arrive at that final solution but it doesn't make sense to have this intermediate solution, I'll do it directly with the full config and we can start using it only for this vars, and then continue with the rest